cwltool/command_line_tool.py

```diff
@@ -1495,8 +1495,22 @@ def collect_output(
                             )
                         primary["format"] = format_eval
                 else:
-                    for primary in aslist(result):
-                        primary["format"] = format_field
+
+                    def recursively_insert(j_dict: Any, key: Any, val: Any) -> Any:
+                        """This is a helper function that recursively inserts
+                            a value into any dictionary."""
```

Comment:
```suggestion
                    def recursively_insert(j_dict: Any, key: Any, val: Any) -> Any:
                        """Recursively insert a value into any dictionary."""
```

Reply:
I exactly had the previous description. The CI complained that it needs to be a proper sentence.


Can you also help me with creating a an empty file as part of the test : 


[https://github.com/common-workflow-language/cwltool/actions/runs/9699693820/job/26770124937#step:8:48](permission denied in CI environment)

Reply:
Hey, sorry for not replying earlier. When you edit a comment, others don't get an updated notification :-)

I left some comments on your PR with hints on how to fix the permission denied issue for the "No leftovers" test